include/libtorrent/announce_entry.hpp

```diff
@@ -37,12 +37,41 @@ POSSIBILITY OF SUCH DAMAGE.
 #include "libtorrent/time.hpp"
 #include "libtorrent/error_code.hpp"
 #include "libtorrent/string_view.hpp"
+#include "libtorrent/address.hpp"
 
 #include <string>
 #include <cstdint>
+#include <vector>
 
 namespace libtorrent
 {
+	namespace aux { struct session_listen_socket; }
+
+	struct TORRENT_EXPORT announce_endpoint
+	{
+		explicit announce_endpoint(aux::session_listen_socket* s)
+			: socket(s), fails(0), updating(false) {}
+		//announce_endpoint(announce_endpoint const&) = default;
+		//announce_endpoint& operator=(announce_endpoint const&) = default;
+
+		// if this tracker has returned an error or warning message
+		// that message is stored here
+		std::string message;
+
+		// if this tracker failed the last time it was contacted
+		// this error code specifies what error occurred
+		error_code last_error;
+
+		aux::session_listen_socket* socket;
```

Comment:
assuming this isn't meant to be public, you could derive an internal class from the public one